### PR TITLE
TE-16.1: basic encap test

### PR DESCRIPTION
**Pull request by singh-prem**
This PR implements TE-16.1 basic encap test as described in https://github.com/openconfig/featureprofiles/pull/2207

### CODE REVIEW

**Comment by NishaSadhasivam-work** [comment]
Please update the static analysis failure on line 703:
func TtlCopyToTunnelHeaderUnsupported should be TTLCopyToTunnelHeaderUnsupported

**Review by NishaSadhasivam-work — DISMISSED** [review]
Please update the static analysis failure on line 703:
func TtlCopyToTunnelHeaderUnsupported should be TTLCopyToTunnelHeaderUnsupported

File: feature/gribi/otg_tests/basic_encap_test/metadata.textproto [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    vendor: CISCO
+  }
+  deviations: {
+    ipv4_missing_enabled: true
+    pf_default_rule_variable_sequence_unsupported: true
+    ttl_copy_to_tunnel_header_unsupported: true
+    gribi_mac_override_with_static_arp: true
+    interface_ref_interface_id_format: true
+  }
+}
```
Comment (Ankur19) [inline feature/gribi/otg_tests/basic_encap_test/metadata.textproto:12]: Add `tags: TAGS_DATACENTER_EDGE` here and a new line

**Comment by xw-g** [comment]
@nflath, can you help review if this PR has any concerns in terms of what expected? If not, we can merge it now and iterate it with other PRs. Thanks.

**Review by nflath — DISMISSED** [review]

**Comment by xw-g** [comment]
@singh-prem can you help resolve the conflict? and then we are good to merge! Thanks!

**Comment by AmrNJ** [comment]
Please can you rebase?

**Comment by xw-g** [comment]
Can you also help fix the pull check ? Thanks.

**Comment by arvbaska1** [comment]
> Can you also help fix the pull check ? Thanks.

hi @xw-g , i am resolving conflicts for this PR . The checkstylecheck is failing since https://github.com/openconfig/featureprofiles/blob/main/feature/gribi/basic_encap/README.md?plain=1#L254 is not defined in contributing.md. 
Should we change the IPv6 address in readme to use 2001:DB8:1::/64 e.g ? or we need to update the contributing.md file ?

File: feature/gribi/otg_tests/basic_encap_test/metadata.textproto (hunk 1 above)
Comment (arvbaska1) [reply feature/gribi/otg_tests/basic_encap_test/metadata.textproto]: updated

**Comment by xw-g** [comment]
> > Can you also help fix the pull check ? Thanks.
> 
> hi @xw-g , i am resolving conflicts for this PR . The checkstylecheck is failing since https://github.com/openconfig/featureprofiles/blob/main/feature/gribi/basic_encap/README.md?plain=1#L254 is not defined in contributing.md. Should we change the IPv6 address in readme to use 2001:DB8:1::/64 e.g ? or we need to update the contributing.md file ?

let's use the IPv6 range defined in the CONTRIBUTION guide. Thanks! Can you help update the README in this PR too? Thanks!

**Comment by arvbaska1** [comment]
> > > Can you also help fix the pull check ? Thanks.
> > 
> > 
> > hi @xw-g , i am resolving conflicts for this PR . The checkstylecheck is failing since https://github.com/openconfig/featureprofiles/blob/main/feature/gribi/basic_encap/README.md?plain=1#L254 is not defined in contributing.md. Should we change the IPv6 address in readme to use 2001:DB8:1::/64 e.g ? or we need to update the contributing.md file ?
> 
> let's use the IPv6 range defined in the CONTRIBUTION guide. Thanks! Can you help update the README in this PR too? Thanks!

Hi @xw-g , sure i can update it , but the check_ip workflow is failing for IPs used in many of the other tests (TE-16.2,TE-17.1,TE-14.2) which are already merged, https://github.com/openconfig/featureprofiles/actions/runs/8085074052/job/22091817406?pr=2456,
so in this PR do we need to change the IPv4 address e.g 138.x as well ?

**Review by nflath — APPROVED** [review]

**Comment by xw-g** [comment]
> > > > Can you also help fix the pull check ? Thanks.
> > > 
> > > 
> > > hi @xw-g , i am resolving conflicts for this PR . The checkstylecheck is failing since https://github.com/openconfig/featureprofiles/blob/main/feature/gribi/basic_encap/README.md?plain=1#L254 is not defined in contributing.md. Should we change the IPv6 address in readme to use 2001:DB8:1::/64 e.g ? or we need to update the contributing.md file ?
> > 
> > 
> > let's use the IPv6 range defined in the CONTRIBUTION guide. Thanks! Can you help update the README in this PR too? Thanks!
> 
> Hi @xw-g , sure i can update it , but the check_ip workflow is failing for IPs used in many of the other tests (TE-16.2,TE-17.1,TE-14.2) which are already merged, https://github.com/openconfig/featureprofiles/actions/runs/8085074052/job/22091817406?pr=2456, so in this PR do we need to change the IPv4 address e.g 138.x as well ?

Thanks, @arvbaska1, you are not required to change the IPs in other test. (FYI, we just changed the workflow so that it only checks for changed files, instead of all the files)

I'm going to merge this PR for now. Can you help file another PR to update the IPv4 and v6 of this TE-16.1 test per the CONTRIBUTION guide? Thanks.

**Comment by arvbaska1** [comment]
> > > > > Can you also help fix the pull check ? Thanks.
> > > > 
> > > > 
> > > > hi @xw-g , i am resolving conflicts for this PR . The checkstylecheck is failing since https://github.com/openconfig/featureprofiles/blob/main/feature/gribi/basic_encap/README.md?plain=1#L254 is not defined in contributing.md. Should we change the IPv6 address in readme to use 2001:DB8:1::/64 e.g ? or we need to update the contributing.md file ?
> > > 
> > > 
> > > let's use the IPv6 range defined in the CONTRIBUTION guide. Thanks! Can you help update the README in this PR too? Thanks!
> > 
> > 
> > Hi @xw-g , sure i can update it , but the check_ip workflow is failing for IPs used in many of the other tests (TE-16.2,TE-17.1,TE-14.2) which are already merged, https://github.com/openconfig/featureprofiles/actions/runs/8085074052/job/22091817406?pr=2456, so in this PR do we need to change the IPv4 address e.g 138.x as well ?
> 
> Thanks, @arvbaska1, you are not required to change the IPs in other test. (FYI, we just changed the workflow so that it only checks for changed files, instead of all the files)
> 
> I'm going to merge this PR for now. Can you help file another PR to update the IPv4 and v6 of this TE-16.1 test per the CONTRIBUTION guide? Thanks.

Sure, i can update them as required based on address range in CONTRIBUTING.md guide

**Pull request merged**